### DISCUSSION TITLE
Is there a way to query InternetServerList with SK2?

### DISCUSSION THREAD

**Discussion by Civ01** [SteamRE/SteamKit#1173] [General]
Talking about:
https://partner.steamgames.com/doc/api/ISteamMatchmakingServers#RequestInternetServerList

**Comment by Civ01** [comment]
So I have created custom Handler, created Method and Callback for it but from what I see I get no response to my request:
```csharp
 public AsyncJob<GetServerListCallback> GetServerList(uint appId)
{
    var msg = new ClientMsgProtobuf<CGameServers_GetServerList_Request>(EMsg.ClientGMSServerQuery);
   //...
    return new AsyncJob<GetServerListCallback>(this.Client, msg.SourceJobID);
}
```
It just dead silence. if I try to `await` `AsyncJob` it throws.

**Comment by yaakov-h** [comment]
If this is available over the CM connection, it will be via `SteamUnifiedMessages` with the `IGameServers` interface.

**Reply by Civ01** [reply]
How can I make/establish this `CM connection`

**Reply by yaakov-h** [reply]
For SK2 that would be the normal `SteamClient` class.
